src/Workspaces/CSharp/Portable/LanguageServices/CSharpSyntaxFactsService.cs

```diff
@@ -317,6 +317,40 @@ public bool IsBaseConstructorInitializer(SyntaxToken token)
         public bool IsQueryExpression(SyntaxNode node)
             => node.Kind() == SyntaxKind.QueryExpression;
 
+        public bool IsQueryKeyword(SyntaxToken token)
+        {
+            switch (token.Kind())
+            {
+                case SyntaxKind.FromKeyword:
+                case SyntaxKind.GroupKeyword:
+                case SyntaxKind.JoinKeyword:
+                case SyntaxKind.IntoKeyword:
+                case SyntaxKind.LetKeyword:
+                case SyntaxKind.ByKeyword:
+                case SyntaxKind.SelectKeyword:
+                case SyntaxKind.OrderByKeyword:
+                case SyntaxKind.OnKeyword:
+                case SyntaxKind.EqualsKeyword:
+                case SyntaxKind.AscendingKeyword:
+                case SyntaxKind.DescendingKeyword:
+                    return true;
```

Comment:
this is a bit dangerous.  it presumes all those keywords will only ever be in query clauses.  Things like "equals" may show up in patterns, and "let" may be allowed in a declaration.

Reply:
This also already merged (and corrected in fce28d03dd6872b283757f092fe2ae7eabc95bb9)

Reply:
I don't see how it's corrected, even in what you linked, there's no verification that these tokens belong to queries.  At the very least, these methods should test that token.Parent is a QueryClause.  

Also, even if merged, we should still do the right thing with code :)

Reply:
Finally, if it's merged... why is it showing up as different?  Does master not have your original chnages?

Reply:
> I don't see how it's corrected

In fce28d0 I went to all the keywords and checked if they can be used in other contexts. I found several other contexts for VB but only `where` was missing for C#. I don't see how `let` and `equals` gone be keywords in other contexts right now.

Reply:
> Finally, if it's merged... why is it showing up as different?

I don't have a clue (I tried for several hours to get this in a clean state but ended either having a good foundation for a PR but being unable to test my changes in VS or with this strange history). My ide was to wait until VS 15.7 get RTM and then either start from scratch or cherry-pick some commits from here. This whole thing was created to:

* See if it can get done as I planned (it did)
* Produce some screenshots (so the IDE team has something to discussed about)
* Get feedback of the changes to the quick info infrastructure I made.

It can be discarded once the points above are settled.